packages/mui-base/src/ButtonUnstyled/ButtonUnstyled.types.ts

```diff
@@ -43,11 +43,7 @@ export interface ButtonUnstyledSlots {
   root?: React.ElementType;
 }
 
-export type ButtonUnstyledProps<
-  D extends React.ElementType = ButtonUnstyledTypeMap['defaultComponent'],
-> = OverrideProps<ButtonUnstyledTypeMap<{}, D>, D> & {
-  component?: D;
-};
+export type ButtonUnstyledProps = DefaultComponentProps<ButtonUnstyledTypeMap>;
```

Comment:
I don't think we need `TypeMap` interfaces anymore. It could be simpler to use just the props types directly.

Reply:
Could we just explicitly define `ButtonUnstyledProps` as `ButtonUnstyledOwnProps & React.ButtonHTMLAttributes<HTMLButtonElement>`? This way readers see at first glance what are the props and we don't need to use the expensive `React.ComponentPropsWithRef` that's in `DefaultComponentProps`

Reply:
Also it makes me wonder if accepting all the native button props is correct in this case. Since the component is not polymorphic, supplying an `a` to `slots.root` while still using some button-specific attributes on the parent could lead to errors. Shouldn't we accept only the common HTML attributes?

Reply:
> Shouldn't we accept only the common HTML attributes?

I think this is too strict. If we go this way, we would need to then add and handle all of the props from the `ButtonHTMLAttributes` interface inside the component:

```ts
autoFocus?: boolean | undefined;
disabled?: boolean | undefined;
form?: string | undefined;
formAction?: string | undefined;
formEncType?: string | undefined;
formMethod?: string | undefined;
formNoValidate?: boolean | undefined;
formTarget?: string | undefined;
name?: string | undefined;
type?: 'submit' | 'reset' | 'button' | undefined;
value?: string | ReadonlyArray<string> | number | undefined;
```

I would go with your first proposal: `ButtonUnstyledOwnProps & React.ButtonHTMLAttributes<HTMLButtonElement>`, and if your second concern becomes a problem (for e.g. we see issues being created about it), we can add runtime warning for using these props - maybe after the component mounts we can check the host element and warn if it is not a `button` and some of the button specific attributes is used.

Reply:
In this case it could be worth creating a LinkButton component and removing the logic to render an `<a>` when `href` is present. I think the Button may be the only component used as frequently with the default component as with the `a` tag.

The props could also be typed as `(ButtonUnstyledWithoutHrefProps &  React.ButtonHTMLAttributes<HTMLButtonElement>) | (ButtonUnstyledWithHrefProps & React.AnchorHTMLAttributes<HTMLAnchorElement>)` - this way the typings will be correct for the most popular cases.

Reply:
Let's not make this discussion a blocker for other components, though. I think this issue is present only in a ButtonUnstyled (and perhaps in InputUnstyled to some extent). All the other components should work just fine with the default component props.

Reply:
This was basically solved with the generic component, I will create an issue for adding a demo for the ButtonUnstyled component used as a Link.